File: Exercises/C#/Assignment2.cs
namespace TestLoop
{
    using System;
    class Program
    {
        static void Main(string[] args)
        {
            /*
            A status report is needed of all government employees.  Statuses are:
            1: Alive, 2: Zombie, 3: Dead, 4: Unknown
            Given an array of int variable, write loops with if else statements and console out everyone’s status.
            Use all loop types.
            Given another array of string variables with names, write out the name and their status.
             */
            int counter = 0;
            int[] statuses = {1, 4, 3, 2, 3, 2};
            string[] names = {"Bob", "Tim", "Tom", "Barb", "Kent", "Carlos"};
            while(counter < statuses.Length)
            {
                Console.Write(names[counter] + " ");
                //switch statement
                switch (statuses[counter])
                {
                    case 1:
                        Console.WriteLine("Status: Alive");
                        break;
                    case 2:
                        Console.WriteLine("Status: Zombie");
                        break;
                    case 3:
                        Console.WriteLine ("Status: Dead");
                        break;
                    case 4:
                        Console.WriteLine("Status: Unknown");
                        break;
                }
                counter++;
            }

            counter = 0;
            do
            {
                Console.Write(names[counter] + " ");
                //switch statement
                switch (statuses[counter])
                {
                    case 1:
                        Console.WriteLine("Status: Alive");
                        break;
                    case 2:
                        Console.WriteLine("Status: Zombie");
                        break;
                    case 3:
                        Console.WriteLine ("Status: Dead");
                        break;
                    case 4:
                        Console.WriteLine("Status: Unknown");
                        break;
                }
                counter++;
            } while(counter < statuses.Length);

            for(counter = 0; counter < statuses.Length; counter++)
            {
                Console.Write(names[counter] + " ");
                //switch statement
                switch (statuses[counter])
                {
                    case 1:
                        Console.WriteLine("Status: Alive");
                        break;
                    case 2:
                        Console.WriteLine("Status: Zombie");
                        break;
                    case 3:
                        Console.WriteLine ("Status: Dead");
                        break;
                    case 4:
                        Console.WriteLine("Status: Unknown");
                        break;
                }
            }

            counter = 0;
            foreach(int status in statuses)
            {
                Console.Write(names[counter] + " ");
                //switch statement
                switch (status)
                {
                    case 1:
                        Console.WriteLine("Status: Alive");
                        break;
                    case 2:
                        Console.WriteLine("Status: Zombie");
                        break;
                    case 3:
                        Console.WriteLine ("Status: Dead");
                        break;
                    case 4:
                        Console.WriteLine("Status: Unknown");
                        break;
                }
                counter++;
            }
        }
    }
}

File: Exercises/JavaScript/Assignment2.js
let statuses = [3,4,2,1,4,2,3];
let names = ['Tim','Tom','Bob','Joe','Jake','John','Steve'];
console.log('for loop');
for(count = 0; count < statuses.length; count++){
    let status = statuses[count];
    let name = names[count];
    console.log(name + ':');
    switch(status){
        case 1:
            console.log('Alive');
            break;
        case 2:
            console.log('Zombie');
            break;
        case 3:
            console.log('Dead');
            break;
        case 4:
            console.log('Unknown');
            break;
    }
}

count = 0;
console.log('while loop');
while(count < statuses.length){
    let status = statuses[count];
    let name = names[count];
    console.log(name + ':');
    switch(status){
        case 1:
            console.log('Alive');
            break;
        case 2:
            console.log('Zombie');
            break;
        case 3:
            console.log('Dead');
            break;
        case 4:
            console.log('Unknown');
            break;
    }
    count++;
}

count = 0;
console.log('do while loop');
do{
    let status = statuses[count];
    let name = names[count];
    console.log(name + ':');
    switch(status){
        case 1:
            console.log('Alive');
            break;
        case 2:
            console.log('Zombie');
            break;
        case 3:
            console.log('Dead');
            break;
        case 4:
            console.log('Unknown');
            break;
    }
    count++;
} while(count < statuses.length)

let personStatuses = [
    {status:4,name:'Tim'},
    {status:2,name:'Tom'},
    {status:1,name:'Bob'},
    {status:3,name:'John'},
    {status:2,name:'Joe'},
    {status:1,name:'Steve'}
];
console.log('for of loop');
for(let personStatus of personStatuses){
    console.log(personStatus.name + ':');
    switch(personStatus.status){
        case 1:
            console.log('Alive');
            break;
        case 2:
            console.log('Zombie');
            break;
        case 3:
            console.log('Dead');
            break;
        case 4:
            console.log('Unknown');
            break;
    }
}